Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -474,6 +474,63 @@ FLATTEN ThrowCompletionOr<Value> Value::to_numeric(GlobalObject& global_object)
     return primitive.to_number(global_object);
 }
 
+struct NumberParseResult {
+    StringView literal;
+    u8 base;
+    bool invalid { false };
+};
+
+static Optional<NumberParseResult> parse_number_text(StringView text)
```

Comment:
On second thought I'm less sure about having just one function, but this for sure needs spec comments since it is different in subtle and less subtle ways from "regular" number parsing

Reply:
Maybe string to number parsing should be extracted out into its own function similar to `Value::string_to_bigint`?

Reply:
> Maybe string to number parsing should be extracted out into its own function similar to `Value::string_to_bigint`?

Yeah something like that seems fine, although if we want to share code with the Parser it probably shouldn't be in `Value`

Reply:
Actually maybe that is too hard and it is a syntax which never occurs in source code so keeping it in value is fine. Sorry I'm so indecisive just thinking about reducing some code deduplication